core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -186,23 +194,40 @@ public void handleNameResolutionError(Status error) {
 
     @Override
     public void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateInfo) {
-      if (stateInfo.getState() == SHUTDOWN && stickinessState != null) {
+      Ref<ConnectivityStateInfo> stateInfoRef = getSubchannelStateInfoRef(subchannel);
+      ConnectivityState stateBefore = stateInfoRef.value.getState();
+      if (stateBefore == SHUTDOWN) {
```

Comment:
This should be an error, as there should be no more events after the SC has shut down.

Reply:
The state in the subchannel attribute is now set to `SHUTDOWN` when the EAG is removed in the `handleResolvedAddressGroups` callback (at the same time as `subchannel.shutdown()` is called and that it's removed from this LB's `subchannels` map - see line 220).

This is because from the load balancer's pov it makes more sense to consider it in a non-ready state as soon as the `NameResolver` reports it removed, rather than when a subsequent event is received that the shutdown is complete (i.e. here).

So this is really catching the "normal" `SHUTDOWN` state change following the subchannel's prior removal.